src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -4787,4 +4752,131 @@ public static StorageCompatibilityMode getStorageCompatibilityMode()
         else
             return conf.storage_compatibility_mode;
     }
+
+    public static <T> ListenableProperty.Remover addBeforeChangePropertyListener(String name, Class<T> clazz, BiConsumer<T, T> consumer)
+    {
+        return addPropertyListener(name, clazz, p -> p.addListener(new ListenableProperty.Listener<Config, T>()
+        {
+            @Override
+            public T before(Config source, String name, T oldValue, T newValue)
+            {
+                consumer.accept(oldValue, newValue);
+                return newValue;
+            }
+        }));
+    }
+
+    public static <T> ListenableProperty.Remover addAfterChangePropertyListener(String name, Class<T> clazz, BiConsumer<T, T> consumer)
+    {
+        return addPropertyListener(name, clazz, p -> p.addListener(new ListenableProperty.Listener<Config, T>()
+        {
+            @Override
+            public void after(Config source, String name, T oldValue, T newValue)
+            {
+                consumer.accept(oldValue, newValue);
+            }
+        }));
+    }
+
+    @SuppressWarnings("unchecked")
+    private static <T> ListenableProperty.Remover addPropertyListener(String name,
+                                                                      Class<T> clazz,
+                                                                      Function<ListenableProperty<Config, T>, ListenableProperty.Remover> adder)
+    {
+        Property property = getPropertyAccessor(name);
+        // The case when replacement property is used and match the property's name.
+        Property property0 = property instanceof Replacement.ReplacementProperty ?
```

Comment:
`property0`? :) Come on ... we can do better.

Reply:
Fixed.